### DISCUSSION TITLE
Register as a DataProvider in Windows

### DISCUSSION THREAD

**Discussion by vsederburg** [mysql-net/MySqlConnector#1186] [Q&A] [answered]
I'm trying to use this with Telerik Reporting.  It works great with everything else, but Telerik expects a dataprovider to be registered on the machine, and without it, it can't seem to find it.

I'm not sure what exactly to ask for, but I know I can't register the dll because it doesn't have the entry points windows expects for regsrv32 to work.  

Is there any way to register MySqlConnector so the machine reports it as a .net data provider?

**Comment by bgrainger** [comment]
My guess would be it's expecting to find a DbProviderFactory registered in `machine.config`. You could install it manually by editing one or both of these files:

* C:\Windows\Microsoft.NET\Framework\v4.0.30319\Config\machine.config
* C:\Windows\Microsoft.NET\Framework64\v4.0.30319\Config\machine.config

You'll need to add the XML block shown here: https://mysqlconnector.net/overview/dbproviderfactories/

There's probably an existing `<system.data>` block you can add it to.

There's no path (to the assembly) specified in `machine.config`, so I'm guessing that means .NET expects to find it in the GAC so you'll have to use `gacutil` to install it.

**Comment by bgrainger** [comment, marked as answer]
> Telerik expects a dataprovider to be registered on the machine

Are you sure it has to be on the machine? You may be able to register the DbProviderFactory locally, just for your app, in app.config (or web.config), following most of the same steps above (but no GAC necessary).

**Reply by vsederburg** [reply]
Yes, that turned out to be it. I just figured it out about 15 minutes ago at last.    
I had added it manually to machine.config, but really did not want to have to figure out how to make an installer do that on a client machine.  Finally removed it from machine.config and tried adding the system.data/DbProviderFactories entry to app.config (exe level) and also app.config in the telerik reporting project, and that seems to have solved the issue.

Thank you for replying!
